core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -146,21 +149,21 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal
     // Note that we assume that all messages have the same persistenceId, which is
     // the case for Akka 2.4.2.
 
-    def serialize(aw: Seq[(PersistentRepr, UUID)]): SerializedAtomicWrite = {
-
-      SerializedAtomicWrite(
-        aw.head._1.persistenceId,
-        aw.map {
-          case (pr, uuid) =>
-            val (pr2, tags) = pr.payload match {
-              case Tagged(payload, ts) =>
-                (pr.withPayload(payload), ts)
-              case _ =>
-                (pr, Set.empty[String])
-            }
-            serializeEvent(pr2, tags, uuid, config.bucketSize, serialization, transportInformation)
-        }
-      )
+    def serialize(aw: Seq[(PersistentRepr, UUID)]): Future[SerializedAtomicWrite] = {
+      val serializedEventsFut: Future[Seq[Serialized]] = Future.sequence(aw.map {
```

Comment:
Using sequence will mean that the serialization will happen in parallel now for an AtomicWrite, i guess this may be a good thing.

Reply:
yes, that is a difference, but I thought it wouldn't be too bad, it's limited by the plugin dispatcher which by default has 0.5 * cores

Reply:
:+1: